sinonjs/nise
# Use @sinonjs/eslint-config

This PR upgrades ~`eslint-config-sinon`~ `@sinonjs/eslint-config` and fixes all uncovered lint violations.

#### Purpose

Adopt the latest `@sinonjs/eslint-config` in order to allow modern JavaScript syntax and ease the maintenance burden of keeping all the development tools up to date.

#### Background

https://hackmd.io/@mroderick/sinon-rfc-drop-legacy-browsers


#### Solution

* Add latest version of `@sinonjs/eslint-config` as `devDependency`
* Fix all lint violations
* Uninstall no longer needed linting dependencies

#### How to verify

1. Check out this branch
2. `npm ci`
3. `npm run lint`
4. Observe that there are no lint errors, only warnings

#### Checklist for author
